### PR TITLE
chore(aws-events-targets) add fargate platform version property

### PR DESCRIPTION
**Pull request by s2-ctraut**
### Commit Message
feat(aws-events-targets) add fargate platform version property

.
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 1]
```diff
@@ -14,6 +14,11 @@ export interface EcsTaskProps {
    */
   readonly cluster: ecs.ICluster;
 
+  /**
+   * Version of Fargate platform
+   */
+  readonly fargatePlatformVersion: ecs.FargatePlatformVersion;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:20]: Did you mean to make this optional?

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 2]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
     this.cluster = props.cluster;
     this.taskDefinition = props.taskDefinition;
     this.taskCount = props.taskCount !== undefined ? props.taskCount : 1;
+    this.fargatePlatformVersion =
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:75]: Use the `??` operator please.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 3]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+      props.fargatePlatformVersion !== undefined
+        ? props.fargatePlatformVersion
+        : ecs.FargatePlatformVersion.LATEST;
 
     if (this.taskDefinition.networkMode === ecs.NetworkMode.AWS_VPC) {
-      const securityGroup = props.securityGroup || this.taskDefinition.node.tryFindChild('SecurityGroup') as ec2.ISecurityGroup;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:71]: Please don't have your IDE reformat unrelated code.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 4]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+          vpc: this.props.cluster.vpc
+        });
     }
   }
 
   /**
    * Allows using tasks as target of CloudWatch events
    */
   public bind(_rule: events.IRule, _id?: string): events.RuleTargetConfig {
-    const policyStatements = [new iam.PolicyStatement({
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:80]: Please don't have your IDE reformat unrelated code.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 5]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+        conditions: {
+          ArnEquals: { 'ecs:cluster': this.cluster.clusterArn }
+        }
+      })
+    ];
 
     // If it so happens that a Task Execution Role was created for the TaskDefinition,
     // then the CloudWatch Events Role must have permissions to pass it (otherwise it doesn't).
     if (this.taskDefinition.executionRole !== undefined) {
-      policyStatements.push(new iam.PolicyStatement({
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:91]: Please don't have your IDE reformat unrelated code.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 6]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+        new iam.PolicyStatement({
+          actions: ['iam:PassRole'],
+          resources: [this.taskDefinition.executionRole.roleArn]
+        })
+      );
     }
 
     // For Fargate task we need permission to pass the task role.
     if (this.taskDefinition.isFargateCompatible) {
-      policyStatements.push(new iam.PolicyStatement({
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:99]: Please don't have your IDE reformat unrelated code.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 7]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+        new iam.PolicyStatement({
+          actions: ['iam:PassRole'],
+          resources: [this.taskDefinition.taskRole.roleArn]
+        })
+      );
     }
 
     const arn = this.cluster.clusterArn;
     const role = singletonEventRole(this.taskDefinition, policyStatements);
-    const containerOverrides = this.props.containerOverrides && this.props.containerOverrides
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:107]: Please don't have your IDE reformat unrelated code.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 8]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+      this.props.containerOverrides &&
+      this.props.containerOverrides.map(({ containerName, ...overrides }) => ({
+        name: containerName,
+        ...overrides
+      }));
     const input = { containerOverrides };
     const taskCount = this.taskCount;
     const taskDefinitionArn = this.taskDefinition.taskDefinitionArn;
 
-    const subnetSelection = this.props.subnetSelection || { subnetType: ec2.SubnetType.PRIVATE };
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:113]: Please don't have your IDE reformat unrelated code.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 9]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+      subnetType: ec2.SubnetType.PRIVATE
+    };
+    const assignPublicIp =
+      subnetSelection.subnetType === ec2.SubnetType.PUBLIC
+        ? 'ENABLED'
+        : 'DISABLED';
 
     const baseEcsParameters = { taskCount, taskDefinitionArn };
 
-    const ecsParameters: events.CfnRule.EcsParametersProperty = this.taskDefinition.networkMode === ecs.NetworkMode.AWS_VPC
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:118]: Please don't have your IDE reformat unrelated code.

File: packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts [hunk 10]
```diff
@@ -66,76 +72,108 @@ export class EcsTask implements events.IRuleTarget {
...
+          }
+        : baseEcsParameters;
 
     return {
       id: '',
       arn,
       role,
       ecsParameters,
       input: events.RuleTargetInput.fromObject(input),
-      targetResource: this.taskDefinition,
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts:138]: Please don't have your IDE reformat unrelated code.

**Comment by cgarvis** [comment]
Closing as this looks to be solved via: #10223
